pydantic_core/_types.py

```diff
@@ -138,6 +138,14 @@ class UnionSchema(TypedDict):
     default: NotRequired[Any]
 
 
+class TupleSchema(TypedDict, total=False):
+    type: Literal['tuple']
+    items: Schema  # default: AnySchema
```

Comment:
IMO it could be good to have `Schema | Sequence[Schema]` to support:
- `tuple[str, ...]` with `items: { type: str}`
- `tuple[str, bool, int]` with `items: [{type: str}, {type: bool}, {type: int}]`
WDYT?

Reply:
@PrettyWood @samuelcolvin I am not 100% sure because I don't know pyright too well, but I believe I stumbled upon an issue with circular (?) / recursive typing and I am struggling to progress.
Is there a Gitter / Discord I could get DM mentoring?

Reply:
I really don't want to add gitter as you can't delete it once created. You can chat to me on Twitter, although I know it's not great for code etc.

Best if you can explain your problem here or in a new issue.

Reply:
Right, so .. I added ` | Sequence[Schema` as suggested to support "rigid" typing of tuple elements, ran the make target and got this:
```pydantic-core/tests/test_typing.py:14:17 - error: Expression of type "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" cannot be assigned to declared type "Schema"
    Type "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" cannot be assigned to type "Schema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "AnySchema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "BoolSchema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "DictSchema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "FloatSchema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "FunctionSchema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "IntSchema"
      "dict[str, str | list[str | dict[str, str | int] | dict[str, str | float] | dict[str, str] | dict[str, str | bool] | dict[str, str | list[int | str]] | dict[str, str | dict[str, str] | int] | dict[str, str | dict[str, str]] | dict[str, str | Type[Foo] | dict[str, str | dict[str, dict[str, str]]]] | dict[str, str | ((bar: str) -> None)] | dict[str, str | dict[str, str | dict[str, dict[str, str] | dict[str, str | list[dict[..., ...]] | None]]]]]]" is incompatible with "ListSchema"
    ... (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations ```

Reply:
I'm afk, I'll look soon.

Reply:
Well, now that I think about it, maybe it's Schema that should now be Union of everything it was + Sequence of itself, rather than tuple `items` field being typed as `Schema | Sequence[Schema]`? 🤔

Reply:
Rather than trying  to explain in comments, I've pushed to your PR with a fix for the types.

Please note, I haven't changed the rust code to reflect these new types. Are you okay to do that?

Let me know if you have any more questions.

Reply:
Thank you, I'll try and find some time in the upcoming days to hopefully finish this :D